### PR TITLE
fix: repack_model writes model data to default bucket if not write permissions

### PR DESCRIPTION
**Pull request by mvsusp**
*Description of changes:*
- fix: repack_model writes model data to default bucket if boto session does not have write permissions to original bucket

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/utils.py [hunk 1]
```diff
@@ -354,7 +356,12 @@ def repack_model(inference_script, source_directory, model_uri, sagemaker_sessio
             bucket, key = url.netloc, url.path.lstrip('/')
             new_key = key.replace(os.path.basename(key), new_model_name)
 
-            sagemaker_session.boto_session.resource('s3').Object(bucket, new_key).upload_file(new_model_path)
+            try:
+                sagemaker_session.boto_session.resource('s3').Object(bucket, new_key).upload_file(new_model_path)
+            except exceptions.S3UploadFailedError:
+                bucket = sagemaker_session.default_bucket()
```
Comment (ChoiByungWook) [inline src/sagemaker/utils.py]: I think we should be always writing to the default bucket.

Feel free to ask for a second opinion.

Comment (mvsusp) [reply src/sagemaker/utils.py]: I agree with you. Good point.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2cf0de28-d197-4a71-a412-948fd571726a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2921f014-9fbd-46d6-9c1e-2b4a74bbdaba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by ChoiByungWook** [comment]
https://github.com/aws/sagemaker-python-sdk/pull/812#discussion_r287079515

**Pull request closed without merging**